src/layout/layout.rs

```diff
@@ -208,23 +208,10 @@ impl Layout {
     /// Initialize an empty cache with a custom size. The cache is keyed on the layout and area, so
     /// that subsequent calls with the same parameters are faster. The cache is a `LruCache`, and
     /// grows until `cache_size` is reached.
-    ///
-    /// Returns true if the cell's value was set by this call.
-    /// Returns false if the cell's value was not set by this call, this means that another thread
-    /// has set this value or that the cache size is already initialized.
-    ///
-    /// Note that a custom cache size will be set only if this function:
-    /// * is called before [`Layout::split()`] otherwise, the cache size is
-    ///   [`Self::DEFAULT_CACHE_SIZE`].
-    /// * is called for the first time, subsequent calls do not modify the cache size.
-    pub fn init_cache(cache_size: usize) -> bool {
-        LAYOUT_CACHE
-            .with(|c| {
-                c.set(RefCell::new(LruCache::new(
-                    NonZeroUsize::new(cache_size).unwrap(),
-                )))
-            })
-            .is_ok()
+    pub fn init_cache(cache_size: usize) {
+        LAYOUT_CACHE.with(|c| {
+            *c.borrow_mut() = LruCache::new(NonZeroUsize::new(cache_size).unwrap());
+        });
```

Comment:
This can be simplified to `LAYOUT_CACHE.set()`, or alternatively with_borrow() and then calling [resize()](https://docs.rs/lru/latest/lru/struct.LruCache.html#method.resize) on the cache. Not sure which would be better. I suppose the resize approach means that we leave anything already cached in place. I'm likely bikeshedding at this point though.

I think also since we're breaking this already, do you think it would be worth moving the NonZeroUsize to the method signature here. This would make constructing a valid value the app's problem and not ours.

Reply:
No no, this is great. I had looked for `reserve` and `set_capacity` which aren't a thing so I thought you couldn't resize the cache.

> I think also since we're breaking this already, do you think it would be worth moving the NonZeroUsize to the method signature here. This would make constructing a valid value the app's problem and not ours.

Yup, seems better for users to find out zero ain't valid at compile time rather than crashing later.